content/en/blog/_posts/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades/2023-08-28-a-new-alpha-mechanism-for-safer-cluster-upgrades.md

```diff
@@ -0,0 +1,51 @@
+---
+layout: blog
+title: "Kubernetes 1.28: A New (alpha) Mechanism For Safer Cluster Upgrades"
+date: 2023-08-28
+slug: kubernetes-1-28-feature-mixed-version-proxy-alpha
+---
+
+**Author:** Richa Banker (Google)
+
+This blog describes the _mixed version proxy_, a new alpha feature in Kubernetes 1.28. The
+mixed version proxy enables an HTTP request for a resource to be served by the correct API server
+in cases where there are multiple API servers at varied versions in a cluster. For example,
+this is useful during a cluster upgrade, or when you're rolling out the runtime configuration of
+the cluster's control plane.
+
+## What problem does this solve?
+When a cluster undergoes an upgrade, the kube-apiservers existing at different versions in that scenario can serve different sets (groups, versions, resources) of built-in resources. A resource request made in this scenario may be served by any of the available apiservers, potentially resulting in the request ending up at an apiserver that may not be aware of the requested resource; consequently it being served a 404 not found error which is incorrect. Furthermore, incorrect serving of the 404 errors can lead to serious consequences such as namespace deletion being blocked incorrectly or objects being garbage collected mistakenly.
+
+## How do we solve the problem?
+
+{{< figure src="mvp-flow-diagram.svg" caption="Mixed Version Proxy Flow Diagram" link="https://mermaid-js.github.io/mermaid-live-editor/edit/?gist=mvp-flow-diagram#pako:eNqVVFFr2zAQ_iuHRzCUOKXpugczCstW9raNrezJDz3Ll1iLI3mS3NWE_PedLMdJmrTLDAb79H33fXeHbh0JXVCURqPRWirpUljHrqQVxSnEOVqKxxACP9FIzCuyfLKGuDZyhaadaVOQ-agrbTzjzXw-94wcxXJhdKMKH_1TStcl6kk9vEfHSju6pyd3InyQfsjenSwXQ_h6-k7QzZ7APg3SeGGojWGz2YxGmcqUpd8NKUGfJC4MrjIF_KBw2kBjyYT_Go2TQtaoHHDwkcwHQAslqqKSapFgLZMQfwk_8_iayBxhvUpye9unTSFEfVWgOdKdj7eiKVzA1QQ-391Djk6Ul49Xl790bi96GlYOpjgBQ1Y3RhBI2xsoIG9fNuyfXuPYy6Gf4ZB12EqXkliQ22gdVFpgVbWBSJUlmOaHdr58vQ85_t_R7FVH4wHEkuzsm9FP7WDMacBuAOBKdPA-NyGphYeDTj6cqzCdnDIPuiZlWcoLCK0UCSe12snvYxvL5A4JSVJ7u4moJCmXCDIOOPvJwyW1XZxzWhKNIS6JQLAbRdW59q9P2-dXziV3pRP2QW_R27Gg5yH6vAweY0drPdSvhwDDBUrFnWf3niAwmcuKnk93dt50g-m3k0F8aOHW6_cw6JKQbztg48qu68FZb_yoxUwvQCqwjup-qB3yuKRtNf1AzL7akPJfJb56pWbj3dXKJzcTKNBhf5FUET52N8KvhdevQ0Bcb_NE42hFZoWy4AW_9sQs6nZ5FnGeqECzzKJMbRjX1Eyhu0LyFoxSZxoaR9xQ_aNVYvsfMP3WDMHNX1rEF5g" >}}
```

Comment:
This figure doesn't render right. You can use the steps in https://kubernetes.io/docs/contribute/new-content/open-a-pr/#preview-locally to check.

Reply:
the [diagram guide](https://kubernetes.io/docs/contribute/style/diagram-guide/) was super helpful, thanks for sharing the link. Apologies for not testing the rendering locally earlier. Fixed it now.